db/changes.go

```diff
@@ -533,12 +533,17 @@ func (db *Database) MultiChangesFeed(ctx context.Context, chans base.Set, option
 
 }
 
-func (db *Database) startChangeWaiter(chans base.Set) *ChangeWaiter {
+func (db *Database) startChangeWaiter(chans channels.Set) (*ChangeWaiter, error) {
 	waitChans := chans
-	if db.user != nil {
-		waitChans = db.user.ExpandWildCardChannel(chans)
+	// FIXME, put collection IDs in a set and expandwildcardchannel on each collectionID
+	//if db.user != nil {
+	//	waitChans = db.user.ExpandWildCardChannel(chans)
+	//}
+	collectionID, err := db.getCollectionID()
+	if err != nil {
+		return nil, err
 	}
-	return db.mutationListener.NewWaiterWithChannels(waitChans, db.user)
+	return db.mutationListener.NewWaiterWithChannels(waitChans, db.user, collectionID), nil
```

Comment:
ChangeWaiter is really just doing string matching, and I think it probably makes sense to have it continue to do that?  Adding a stringify to channels.ID might be the simplest way to do this - I don't think there are any collisions if you just do basic concatenation.

Reply:
You could still change NewWaiterWithChannels to take channels.Set as a parameter (and do the string generation in that method), but things like Notify could be left as-is.  Adding a new NotifyChannels might be reasonable if you want to pass in a channels.Set to avoid converting a channels.Set to a base.Set everywhere.